app/assets/javascripts/discourse/app/components/edit-topic-timer-form.js

```diff
@@ -86,26 +86,27 @@ export default Component.extend({
 
   @discourseComputed()
   customTimeShortcutOptions() {
+    const timezone = this.currentUser.resolvedTimezone(this.currentUser);
```

Comment:
That's a weird api we have there! (it's to handle the updates of current user's saved timezone, but that logic should probably be moved elsewhere 🤔)

Reply:
Yes, I agree. I wanted to change it once, thought it was going to be easy and fast. But there was something tricky about changing this. Going to have another look at it later.

Reply:
It's like this so that any users without a timezone have it set + saved when we try and get it, otherwise it would end up nil. However, on both the email_login and login routes we set the timezone for the user, so maybe this is a rare case?

Maybe we just change it to this:

```javascript
timezone() {
  if (!this._timezone) {
    return moment.tz.guess();
  }
  return this._timezone;
},
```

But, that's the job for another PR :)